flux-desugar/src/desugar.rs

```diff
@@ -429,9 +429,14 @@ impl ParamsCtxt<'_> {
             }
             surface::TyKind::Constr(_, ty) => self.ty_gather_params(ty, adt_sorts),
 
-            surface::TyKind::Path(_) | surface::TyKind::Exists { .. } | surface::TyKind::Unit => {
+            surface::TyKind::Path(path) => {
+                for ty in &path.args {
+                    let _ = self.ty_gather_params(ty, adt_sorts);
```

Comment:
So what's your take about the following definition now?

```rust
fn foo(Vec<i32[@n]>) {}
```

should we fail at the call site (fail to infer parameter) or at the definition site (you can't have a binder there). If we agree on the latter, maybe we can leave a TODO for adding a check somewhere to catch this situation.

Reply:
I added the code to check for the illegal binder at the definition site -- so now the above should be rejected (see test `param_infer00.rs`)